reporter/src/main/kotlin/model/AmazonOssAttributionBuilderService.kt

```diff
@@ -0,0 +1,207 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonValue
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+import okhttp3.OkHttpClient
+
+import retrofit2.Call
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.POST
+import retrofit2.http.Path
+
+/**
+ * A project's ACL is a map of group names to roles, see
+ * https://github.com/amzn/oss-attribution-builder/issues/31#issuecomment-568004235
+ * and
+ * https://github.com/amzn/oss-attribution-builder/blob/8eda70c/docs/openapi.yaml#L89-L96.
+ */
+typealias ProjectAcl = Map<String, AmazonOssAttributionBuilderService.Role>
+
+/**
+ * Interface for the Amazon OSS Attribution Builder REST API, based on code generated by https://app.quicktype.io/
+ * from https://github.com/amzn/oss-attribution-builder/blob/master/docs/openapi.yaml.
+ */
+interface AmazonOssAttributionBuilderService {
+    companion object {
+        /**
+         * Create an OSS Attribution Builder service instance for communicating with the given [server], optionally
+         * using a pre-built OkHttp [client].
+         */
+        fun create(server: Server, client: OkHttpClient? = null): AmazonOssAttributionBuilderService {
+            val retrofit = Retrofit.Builder()
+                .apply { if (client != null) client(client) }
+                .baseUrl(server.url)
+                .addConverterFactory(JacksonConverterFactory.create(JsonMapper().registerKotlinModule()))
+                .build()
+
+            return retrofit.create(AmazonOssAttributionBuilderService::class.java)
+        }
+    }
+
+    /**
+     * See https://github.com/amzn/oss-attribution-builder/blob/8eda70c/docs/openapi.yaml#L7-L9.
+     */
+    enum class Server(val url: String) {
+        DEFAULT("http://localhost:8000/api/v1/"),
+        DEV("http://localhost:2424/api/v1/")
+    }
+
+    /**
+     * See https://github.com/amzn/oss-attribution-builder/blob/8eda70c/docs/openapi.yaml#L217-L233.
+     */
+    data class NewProject(
```

Comment:
Prefer to not call this `NewProject` but simply `Project` as the project is not new but you submitting it to OSS attribution builder 'projects/new' API endpoint.

Reply:
Well, it *is* new to the OSS Attribution Builder, so in this context I would argue the name to be fitting.